scripts/exgdas_global_marine_analysis_prep.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+################################################################################
+#  UNIX Script Documentation Block
+#                      .                                             .
+# Script name:         exufsda_global_marine_analysis_prep.py
+# Script description:  Stages files and generates YAML for UFS Global Marine Analysis
+#
+# Author: Guillaume Vernieres      Org: NCEP/EMC     Date: 2022-03-28
+#
+# Abstract: This script stages the marine observations necessary
+#           to produce a UFS Global Marine Analysis.
+#
+# $Id$
+#
+# Attributes:
+#   Language: Python3
+#
+################################################################################
+
+# import os and sys to add ush to path
+import os
+import sys
+import yaml
+import glob
+import dateutil.parser as dparser
+
+# get absolute path of ush/ directory either from env or relative to this file
+my_dir = os.path.dirname(__file__)
+my_home = os.path.dirname(os.path.dirname(my_dir))
+ufsda_home = os.path.join(os.getenv('HOMEgfs'), 'sorc', 'ufs_da.fd', 'UFS-DA')
+sys.path.append(os.path.join(os.getenv('HOMEgfs', my_home), 'ush'))
+print(f"sys.path={sys.path}")
+
+
+# import UFSDA utilities
+import ufsda
+
+
+def gen_bkg_list(bkg_path='.', file_type='MOM', yaml_name='bkg.yaml'):
+    # generate a YAML of the list the backgrounds for the pseudo model
+    # TODO (Guillaume): Move somewhere else ...
+    files = glob.glob(bkg_path+'/*'+file_type+'*')
+    files.sort()
+    bkg_list = []
+    for bkg in files:
```

Comment:
can SOCA do the date templating that FV3 JEDI can do? If so, this is not needed, right?

Reply:
... Yes, I knew I was forgetting something. I'll test but it should work.

Reply:
@CoryMartin-NOAA , turns out no, I can't do what you are doing in 4D h(x). `PSEUDO` is a factory in [fv3-jedi](https://github.com/JCSDA-internal/fv3-jedi/blob/a553b461ced0bd6810bfef4774889a6bb660f95f/src/fv3jedi/Model/pseudo/ModelPseudo.cc#L24). We should modify`PseudoModel` at some point ...

Reply:
Darn. Yes, OOPS should be smart enough to look for files valid at whatever time step it is requesting based on a templated path.